tests/php/data/type_rendering_constructorsynopsis_parameters-and-return-type.xml

```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0" encoding="utf-8"?>
+<chapter xml:id="type-rendering-constructorsynopsis-parameters-and-return-type">
+
+ <section>
+  <para>1. Constructor with no parameters, no return type</para>
+  <constructorsynopsis>
+   <modifier>final</modifier> <modifier>public</modifier> <methodname>ClassName::__construct</methodname>
+  </constructorsynopsis>
+ </section>
+
+ <section>
+  <para>2. Constructor with no parameters, void (element) return type</para>
+  <constructorsynopsis>
+   <modifier>final</modifier> <modifier>public</modifier> <methodname>ClassName::__construct</methodname>
+   <void/>
+  </constructorsynopsis>
+ </section>
+
+ <section>
+  <para>3. Constructor with one parameter</para>
+  <constructorsynopsis>
+   <modifier>final</modifier> <modifier>private</modifier> <methodname>ClassName::__construct</methodname>
+   <methodparam><type>mixed</type><parameter>anything</parameter></methodparam>
+  </constructorsynopsis>
+ </section>
+
+ <section>
+  <para>4. Constructor with optional parameter</para>
+  <constructorsynopsis>
+   <modifier>final</modifier> <modifier>protected</modifier> <methodname>ClassName::__construct</methodname>
+   <methodparam choice="opt"><type>int</type><parameter>count</parameter><initializer>0</initializer></methodparam>
+  </constructorsynopsis>
+ </section>
+
+ <section>
+  <para>5. Constructor with nullable parameter</para>
+  <constructorsynopsis>
+   <modifier>final</modifier> <modifier>public</modifier> <methodname>ClassName::__construct</methodname>
+   <methodparam><type class="union"><type>float</type><type>null</type></type><parameter>value</parameter></methodparam>
+  </constructorsynopsis>
+ </section>
+
+ <section>
+  <para>6. Constructor with nullable optional parameter</para>
+  <constructorsynopsis>
+   <modifier>final</modifier> <modifier>private</modifier> <methodname>ClassName::__construct</methodname>
+   <methodparam choice="opt"><type class="union"><type>string</type><type>null</type></type><parameter>options</parameter><initializer>""</initializer></methodparam>
+  </constructorsynopsis>
+ </section>
+
+ <section>
+  <para>7. Constructor with reference parameter</para>
+  <constructorsynopsis>
+   <modifier>final</modifier> <modifier>protected</modifier> <methodname>ClassName::__construct</methodname>
+   <methodparam><type>array</type><parameter role="reference">reference</parameter></methodparam>
+  </constructorsynopsis>
+ </section>
+
+ <section>
+  <para>8. Constructor with union type parameter</para>
+  <constructorsynopsis>
+   <modifier>final</modifier> <modifier>public</modifier> <methodname>ClassName::__construct</methodname>
+   <methodparam><type class="union"><type>iterable</type><type>resource</type><type>callable</type><type>null</type></type><parameter>option</parameter></methodparam>
+  </constructorsynopsis>
+ </section>
```

Comment:
I think the "logical" way to have an intersection type in markup would be:
```xml
<type class="intersection"><type>Countable</type><type>Traversable</type></type>
```
and a DNF type would be:
```xml
<type class="union"><type class="intersection"><type>Countable</type><type>Traversable</type></type><type>DOMAttr</type></type>
```

But the rendering with the parenthesis for DNF types might be tricky and could be added later, intersection types tho I would imagine would "just" work.

Reply:
Correct, there are _good_ implementation reasons as to why I forced it to be never more nested than `A|(B&C)` for the moment. :)

Reply:
For the moment? :-(

What would be the correct `xml` output for your DNF example? Is the below correct or am I missing a `<span class="type">` and a `</span>` just inside the parentheses to wrap the intersection type? 
```xml
<span class="type">(<span class="type">Countable</span>&amp;<span class="type">Traversable</span>)|<span class="type">DOMAttr</span></span>
```

Reply:
We *might* need to support generalized type nesting for php-src for type aliases, but I'm not sure that's really something to bother about for the docs.

Looking at my docbook render, I settled on (added some line breaks):
```html
<span class="type">
    (<span class="type">Countable</span>&amp;<span class="type">Traversable</span>)|
    <span class="type">DOMAttr</span>
</span>
```